Mathlib/Tactic/GCongr/Core.lean

```diff
@@ -126,50 +126,72 @@ The `rel` tactic is finishing-only: if fails if any main or side goals are not r
 namespace Mathlib.Tactic.GCongr
 open Lean Meta
 
+/-- `GCongrKey` is the key used to store and look up `gcongr` lemmas. -/
+structure GCongrKey where
+  /-- The name of the relation.
+  For example, `a + b ≤ a + c` has ``relName := `LE.le``. -/
+  relName : Name
+  /-- The name of the head function.
+  For example, `a + b ≤ a + c` has ``head := `HAdd.hAdd``. -/
+  head : Name
+  /-- The array of which arguments in the application of `head` are different.
+  For example, `a + b ≤ a + c` has `#[false, false, false, false, false, true]`.
+  We interpret `∀` and `fun` as having two arguments: the domain and the body. -/
+  varyingArgs : Array Bool
+deriving Inhabited, BEq, Hashable
+
 /-- Structure recording the data for a "generalized congruence" (`gcongr`) lemma. -/
 structure GCongrLemma where
+  /-- The name of the lemma. -/
   declName : Name
-  mainSubgoals : Array (Nat × Nat)
-  varyingArgs : Array Bool
```

Comment:
So you're moving `varyingArgs` out of this structure. Any particular reason? Is this already present in `GCongrKey`?